JoinColony/colonyDapp
# Add Broadcaster to starting script

This PR adds the `braoadcaster` service to the startup orchestration script, as well setting up an example metatransaction saga.

**NOTE:** This adds new stuff both to `colonyJS`, `colonyNetwork`, and the dev orchestration script, so please:
- remove your node modules: `rm -R node_modules` _(this step is just to make sure, as I had issues with this step)_
- reinstall `npm i`
- provision `npm run provision`

### Testing:
- create a colony
- copy that colony's address
- open your redux dev tools, and dispatch this action, replacing the colony address with the one you've copied above: _(same as I do in the demo)_
```js
{
  type: 'META_MINT_TOKENS',
  payload: {
    colonyAddress: '<colonyAddress>',
    amount: '1000000000000000000',
  },
}
```
- _(have a look at the console, as everything is logged out)_
- you should have a new mint tokens action created, without actually firing a transaction _(this is even more visible is using a `Metamask` wallet, since that one will ask you to sign a message, the local ganche wallet, will just do it)_

### Changes:
- [x] refactored the startup script _(for the broadcaster, but for the reputation oracle and miner as well)_
- [x] some fixes related to how the dapp handles chainId's and networks
- [x] added redux action and types for `META_MINT_TOKENS`
- [x] some minor cleanup _(unused packages, leftover constants)_
- [x] added the `createMintTokensMetaAction` saga _(this is just intial wiring, to prove it works, it will get refactored properly in #2779)_

### Demo:
![demo-metatransactions](https://user-images.githubusercontent.com/1193222/138946981-e77a34d0-1c46-4a7a-8726-89976f89c777.gif)
